src/main/java/io/github/phantamanta44/libnine/client/gui/component/GuiComponentManager.java

```diff
@@ -82,9 +106,45 @@ boolean handleMouseDrag(int mX, int mY, int button, long dragTime) {
         }
 
         boolean handleKeyTyped(char typed, int keyCode) {
-            return comp.onKeyPress(keyCode, typed);
+            boolean success = comp.onKeyPress(keyCode, typed);
+            if (keyCode == Keyboard.KEY_ESCAPE
```

Comment:
we don't want component-level keys and gui-level keys to process at the same time. it might be best to just check the gui-level keys first then skip the component-level keys if needed